statistics/histogram.go

```diff
@@ -671,11 +687,13 @@ func MergeHistograms(sc *stmtctx.StatementContext, lh *Histogram, rh *Histogram,
 
 // AvgCountPerValue gets the average row count per value by the data of histogram.
 func (hg *Histogram) AvgCountPerValue(totalCount int64) float64 {
+	factor := hg.getIncreaseFactor(totalCount)
+	totalNotNull := hg.notNullCount() * factor
 	curNDV := float64(hg.NDV) * hg.getIncreaseFactor(totalCount)
 	if curNDV == 0 {
 		curNDV = 1
 	}
-	return float64(totalCount) / curNDV
+	return totalNotNull / curNDV
```

Comment:
`NDV` should not contain `NULL`? /cc @lamxTyler

Reply:
In column histograms, it does not contains NULL. In index histograms, it may contain because we do not specially handle the case of all columns with a NULL value.